Lib/Src/CUartCom.cpp

```diff
@@ -113,14 +129,38 @@ bool CUartCom::send(etl::string<MAX_STRING_SIZE> msg)
 }
 
 /**
- * @brief Update TX Buffer with next element on queue
+ * @brief Add data to the TX Queue and start transmission if possible
+ * @param Pointer to data buffer
+ * @param Length of the data buffer (in bytes)
+ * @return True if message was added to the queue and transmission started
+ * successfully.
  */
-void CUartCom::updateTxBuffer()
+bool CUartCom::send(uint8_t *p_data_buf, uint32_t len)
```

Comment:
Why are there two methods that basically do the same thing?

Reply:
```
bool CUartCom::send(uint8_t *p_data_buf, uint32_t len)
{
    etl::string<max_string_size> tx_string;
    // copy p_data_buf into tx_string
    bool b_success = send(tx_string);
    return b_success;
}
```

Reply:
The actual send logic should only be implemented in one place. Otherwise, you are running the danger of having to non-identical methods. One causing bugs and one not. Bad for maintainability.